### PR TITLE
Add boxing support to IObservable within auto registering types

### PR DESCRIPTION
**Pull request by Shane32**

### CODE REVIEW

File: src/GraphQL.Tests/Types/AutoRegisteringObservableTests.cs [hunk 1]
```diff
@@ -219,9 +243,17 @@ public class TestClass
         public static async Task<IObservable<string>> ListOfStrings2() => new string[] { "a", "b", "c" }.ToObservable();
         public static async ValueTask<IObservable<string>> ListOfStrings3() => new string[] { "a", "b", "c" }.ToObservable();
         [OutputType(typeof(StringGraphType))]
-        public static IObservable<object> ListOfStrings4() => new string[] { "a", "b", "c" }.ToObservable();
+        public static IObservable<object> ListOfStrings4() => new object[] { "a", "b", "c" }.ToObservable();
         [OutputType(typeof(StringGraphType))]
-        public static async ValueTask<IObservable<string>> ListOfStrings5() => new string[] { "a", "b", "c" }.ToObservable();
+        public static async ValueTask<IObservable<object>> ListOfStrings5() => new object[] { "a", "b", "c" }.ToObservable();
```
Comment (Shane32) [inline src/GraphQL.Tests/Types/AutoRegisteringObservableTests.cs:248]: This test was incorrect and has been fixed (ListOfStrings3 already tests this scenario).

Comment (sungam3r) [reply src/GraphQL.Tests/Types/AutoRegisteringObservableTests.cs:248]: One more combination - `Task<IObservable<object>>`

File: src/GraphQL/Resolvers/SourceStreamMethodResolver.cs [hunk 2]
```diff
@@ -149,13 +160,48 @@ public SourceStreamMethodResolver(MethodInfo methodInfo, LambdaExpression instan
...
+            private sealed class ObserverAdapter : IObserver<T>
+            {
+                private readonly IObserver<object?> _observer;
+                public ObserverAdapter(IObserver<object?> observer)
+                {
+                    _observer = observer;
+                }
+                public void OnCompleted() => _observer.OnCompleted();
+                public void OnError(Exception error) => _observer.OnError(error);
+                public void OnNext(T value) => _observer.OnNext(value);  //<--- boxing here
```
Comment (sungam3r) [inline src/GraphQL/Resolvers/SourceStreamMethodResolver.cs:203]:
```suggestion
                public void OnNext(T value) => _observer.OnNext(value); // note: boxing here
```

File: src/GraphQL.Tests/Types/AutoRegisteringObservableTests.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL.Tests/Types/AutoRegisteringObservableTests.cs]: done

**Comment by sungam3r** [comment]
Let's merge after discuss of #3487.

**Comment by Shane32** [comment]
Looks good to me.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**